### PR TITLE
Add documentation for cluster-level coordination setting changes

### LINKED ISSUE #5603: [DOC] Update documentation for cluster settings to include leader/checker timeout setting
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 We recently made leader/checker timeout setting dynamic. The document is to be updated to reflect the changes.
https://github.com/opensearch-project/OpenSearch/issues/10528

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by niyatiagg**
### Description
Adds documentation for the Leader check timeout and follower check timeout to be set dynamically through cluster update API

### Issues Resolved
Closes #5603 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 1]
```diff
@@ -114,8 +114,14 @@ OpenSearch supports the following cluster-level shard, block, and task settings:
...
+- `cluster.persistent_tasks.allocation.recheck_interval` (Time unit): The cluster manager automatically checks whether persistent tasks need to be assigned when the cluster state changes in a significant way. There are other factors, such as memory usage, that will affect whether persistent tasks are assigned to nodes but do not otherwise cause the cluster state to change. This setting defines how often assignment checks are performed in response to these factors. Default is `30 seconds`, with a minimum of `10 seconds` being required.
 
 ## Cluster-level index settings
 
-For information about index-level index settings, see [Cluster-level index settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index-settings/#cluster-level-index-settings).
\ No newline at end of file
+For information about index-level index settings, see [Cluster-level index settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index-settings/#cluster-level-index-settings).
+
+## Cluster-level Coordination settings
```
Comment (kolchfa-aws) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:123]:
```suggestion
## Cluster-level coordination settings
```

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 2]
```diff
@@ -114,8 +114,14 @@ OpenSearch supports the following cluster-level shard, block, and task settings:
...
+
```
Comment (kolchfa-aws) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:124]:
```suggestion

OpenSearch supports the following cluster-level coordination settings. All settings in this list are dynamic:

```

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 3]
```diff
@@ -114,8 +114,14 @@ OpenSearch supports the following cluster-level shard, block, and task settings:
...
+- `cluster.fault_detection.leader_check.timeout` (Time unit): Determines the timeout duration for each node, indicating the period it waits for a response from the cluster manager during a leader check before deeming it as a failure. It can now be set dynamically through Update cluster settings API.
```
Comment (kolchfa-aws) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:125]:
```suggestion
- `cluster.fault_detection.leader_check.timeout` (Time unit): Determines the amount of time a node waits for a response from the cluster manager during a leader check before deeming the check as a failure. 
```

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 4]
```diff
@@ -114,8 +114,14 @@ OpenSearch supports the following cluster-level shard, block, and task settings:
...
+
+- `cluster.fault_detection.follower_check.timeout` (Time unit): Determines the timeout duration for the cluster manager, specifying the duration it waits for a response during a follower check before deeming it as a failure. It can now be set dynamically through Update cluster settings API.
```
Comment (kolchfa-aws) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:127]:
```suggestion
- `cluster.fault_detection.follower_check.timeout` (Time unit): Determines the amount of time the cluster manager waits for a response during a follower check before deeming the check as a failure.
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @niyatiagg! A couple of changes and then we'll move the PR to editorial review.

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 5]
```diff
@@ -114,8 +114,16 @@ OpenSearch supports the following cluster-level shard, block, and task settings:
...
 
-For information about index-level index settings, see [Cluster-level index settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index-settings/#cluster-level-index-settings).
\ No newline at end of file
+For information about index-level index settings, see [Cluster-level index settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index-settings/#cluster-level-index-settings).
+
+## Cluster-level coordination settings
+
+OpenSearch supports the following cluster-level coordination settings. All settings in this list are dynamic:
+
+- `cluster.fault_detection.leader_check.timeout` (Time unit): Determines the amount of time a node waits for a response from the cluster manager during a leader check before deeming the check as a failure.
```
Comment (owaiskazi19) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:127]: We should also mention the default timeout here.

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 6]
```diff
@@ -114,8 +114,16 @@ OpenSearch supports the following cluster-level shard, block, and task settings:
...
\ No newline at end of file
+For information about index-level index settings, see [Cluster-level index settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index-settings/#cluster-level-index-settings).
+
+## Cluster-level coordination settings
+
+OpenSearch supports the following cluster-level coordination settings. All settings in this list are dynamic:
+
+- `cluster.fault_detection.leader_check.timeout` (Time unit): Determines the amount of time a node waits for a response from the cluster manager during a leader check before deeming the check as a failure. 
+
+- `cluster.fault_detection.follower_check.timeout` (Time unit): Determines the amount of time the cluster manager waits for a response during a follower check before deeming the check as a failure.
```
Comment (owaiskazi19) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:129]: Need default timeout here as well.

File: _install-and-configure/configuring-opensearch/cluster-settings.md (hunk 5 above)
Comment (owaiskazi19) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:127]:
```suggestion
- `cluster.fault_detection.leader_check.timeout` (Time unit): Determines the amount of time a node waits for a response from the elected cluster manager during a leader check before deeming the check as a failure. 
```

File: _install-and-configure/configuring-opensearch/cluster-settings.md (hunk 6 above)
Comment (owaiskazi19) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:129]:
```suggestion
- `cluster.fault_detection.follower_check.timeout` (Time unit): Determines the amount of time the elected cluster manager waits for a response during a follower check before deeming the check as a failure.
```

**Review by owaiskazi19 — COMMENTED** [review]
Thanks @niyatiagg for the documentation. LGTM overall with few suggestions. 
@jainankitk can you take another pass? And also do you think we should include a note saying changing the settings can cause the cluster to become unstable?

File: _install-and-configure/configuring-opensearch/cluster-settings.md (hunk 5 above)
Comment (jainankitk) [reply _install-and-configure/configuring-opensearch/cluster-settings.md:127]: +1, let us mention the default and min/max values

File: _install-and-configure/configuring-opensearch/cluster-settings.md (hunk 6 above)
Comment (jainankitk) [reply _install-and-configure/configuring-opensearch/cluster-settings.md:129]: +1, let us mention the default and min/max values

**Comment by jainankitk** [comment]
> Thanks @niyatiagg for the documentation. LGTM overall with few suggestions. @jainankitk can you take another pass? 

Agree with specifying default value.

> And also do you think we should include a note saying changing the settings can cause the cluster to become unstable?

Yeah, that can prevent customers from tweaking this unknowingly. `Changing this setting from the default value can result into unstable cluster.`

File: _install-and-configure/configuring-opensearch/cluster-settings.md (hunk 6 above)
Comment (owaiskazi19) [reply _install-and-configure/configuring-opensearch/cluster-settings.md:129]: Here as well `Changing this setting from the default value may cause an unstable cluster.` last sentence.

File: _install-and-configure/configuring-opensearch/cluster-settings.md (hunk 5 above)
Comment (owaiskazi19) [reply _install-and-configure/configuring-opensearch/cluster-settings.md:127]: @niyatiagg you can include `Changing this setting from the default value may cause an unstable cluster.` as the last sentence here.

**Review by jainankitk — APPROVED** [review]
LGTM!

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 7]
```diff
@@ -114,8 +114,16 @@ OpenSearch supports the following cluster-level shard, block, and task settings:
...
 
-For information about index-level index settings, see [Cluster-level index settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index-settings/#cluster-level-index-settings).
\ No newline at end of file
+For information about index-level index settings, see [Cluster-level index settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index-settings/#cluster-level-index-settings).
+
+## Cluster-level coordination settings
+
+OpenSearch supports the following cluster-level coordination settings. All settings in this list are dynamic:
+
+- `cluster.fault_detection.leader_check.timeout` (Time unit): Determines the amount of time a node waits for a response from the elected cluster manager during a leader check before deeming the check as a failure. Possible value can range from `1ms` to `60s` with default being `10s`. Changing this setting from the default value can result into an unstable cluster.
```
Comment (kolchfa-aws) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:127]:
```suggestion
- `cluster.fault_detection.leader_check.timeout` (Time unit): The amount of time a node waits for a response from the elected cluster manager during a leader check before deeming the check as a failure. Valid values are from `1ms` to `60s`, inclusive. Default is `10s`. Changing this setting to a value other than the default can result in an unstable cluster.
```

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 8]
```diff
@@ -114,8 +114,16 @@ OpenSearch supports the following cluster-level shard, block, and task settings:
...
+
+- `cluster.fault_detection.follower_check.timeout` (Time unit): Determines the amount of time the elected cluster manager waits for a response during a follower check before deeming the check as a failure. Possible value can range from `1ms` to `60s` with default being `10s`. Changing this setting from the default value can result into an unstable cluster.
```
Comment (kolchfa-aws) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:129]:
```suggestion
- `cluster.fault_detection.follower_check.timeout` (Time unit): Determines the amount of time the elected cluster manager waits for a response during a follower check before deeming the check as a failure. Valid values are from `1ms` to `60s`, inclusive. Default is `10s`. Changing this setting to a value other than the default can result in an unstable cluster.
```

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 9]
```diff
@@ -114,8 +114,16 @@ OpenSearch supports the following cluster-level shard, block, and task settings:
...
\ No newline at end of file
+For information about index-level index settings, see [Cluster-level index settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index-settings/#cluster-level-index-settings).
+
+## Cluster-level coordination settings
+
+OpenSearch supports the following cluster-level coordination settings. All settings in this list are dynamic:
+
+- `cluster.fault_detection.leader_check.timeout` (Time unit): The amount of time a node waits for a response from the elected cluster manager during a leader check before deeming the check as a failure. Valid values are from `1ms` to `60s`, inclusive. Default is `10s`. Changing this setting to a value other than the default can result in an unstable cluster.
+
+- `cluster.fault_detection.follower_check.timeout` (Time unit): Determines the amount of time the elected cluster manager waits for a response during a follower check before deeming the check as a failure. Valid values are from `1ms` to `60s`, inclusive. Default is `10s`. Changing this setting to a value other than the default can result in an unstable cluster.
```
Comment (kolchfa-aws) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:129]:
```suggestion
- `cluster.fault_detection.follower_check.timeout` (Time unit): The amount of time the elected cluster manager waits for a response during a follower check before deeming the check as a failure. Valid values are from `1ms` to `60s`, inclusive. Default is `10s`. Changing this setting to a value other than the default can result in an unstable cluster.
```

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 10]
```diff
@@ -114,8 +114,16 @@ OpenSearch supports the following cluster-level shard, block, and task settings:
...
 
-For information about index-level index settings, see [Cluster-level index settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index-settings/#cluster-level-index-settings).
\ No newline at end of file
+For information about index-level index settings, see [Cluster-level index settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index-settings/#cluster-level-index-settings).
+
+## Cluster-level coordination settings
+
+OpenSearch supports the following cluster-level coordination settings. All settings in this list are dynamic:
+
+- `cluster.fault_detection.leader_check.timeout` (Time unit): The amount of time a node waits for a response from the elected cluster manager during a leader check before deeming the check as a failure. Valid values are from `1ms` to `60s`, inclusive. Default is `10s`. Changing this setting to a value other than the default can result in an unstable cluster.
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:127]:
```suggestion
- `cluster.fault_detection.leader_check.timeout` (Time unit): The amount of time a node waits for a response from the elected cluster manager during a leader check before deeming the check a failure. Valid values are from `1ms` to `60s`, inclusive. Default is `10s`. Changing this setting to a value other than the default can result in an unstable cluster.
```

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 11]
```diff
@@ -114,8 +114,16 @@ OpenSearch supports the following cluster-level shard, block, and task settings:
...
+
+- `cluster.fault_detection.follower_check.timeout` (Time unit): The amount of time the elected cluster manager waits for a response during a follower check before deeming the check as a failure. Valid values are from `1ms` to `60s`, inclusive. Default is `10s`. Changing this setting to a value other than the default can result in an unstable cluster.
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:129]:
```suggestion
- `cluster.fault_detection.follower_check.timeout` (Time unit): The amount of time the elected cluster manager waits for a response during a follower check before deeming the check a failure. Valid values are from `1ms` to `60s`, inclusive. Default is `10s`. Changing this setting to a value other than the default can result in an unstable cluster.
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Just a couple of words to remove. Thanks!

**Pull request merged**